File: DekafTestDB/tests/core/CommonExceptionRecognizingTest.java
package org.jetbrains.dekaf.core;

import org.jetbrains.dekaf.CommonIntegrationCase;
import org.jetbrains.dekaf.exceptions.NoTableOrViewException;
import org.junit.Before;
import org.junit.Test;



/**
 * @author Leonid Bushuev from JetBrains
 */
public class CommonExceptionRecognizingTest extends CommonIntegrationCase {

  @Before
  public void setUp() throws Exception {
    DB.connect();
  }

  @Test(expected = NoTableOrViewException.class)
  public void recognize_NoTableOrView() {
    DB.inTransactionDo(tran ->
        tran.query("select * from unexistent_table", Layouts.existence()).run()
    );
  }

}

File: DekafTestDB/tests/core/CommonExceptionRecognizingTest.kt
package org.jetbrains.dekaf.core

import org.jetbrains.dekaf.CommonIntegrationCase
import org.jetbrains.dekaf.assertions.expectedException
import org.jetbrains.dekaf.exceptions.NoTableOrViewException
import org.junit.jupiter.api.BeforeAll
import org.junit.jupiter.api.Test


/**
 * @author Leonid Bushuev from JetBrains
 */
class CommonExceptionRecognizingTest : CommonIntegrationCase() {

    @BeforeAll
    @Throws(Exception::class)
    fun setUp() {
        DB.connect()
    }

    @Test
    fun recognize_NoTableOrView() {
        val queryText = "select * from unexistent_table"
        expectedException<NoTableOrViewException> {
            DB.inTransactionDo { tran -> tran.query(queryText, layoutExistence()).run() }
        }
    }

}
